File: backend-ruby/modules/payments/public_api/payments_public_api.rb
# ONLY interface other modules/apps may import from the payments module.
require_relative "../internal/payment_service"

module Payments
  module PublicApi
    class PaymentsPublicApi
      def initialize
        @svc = Payments::Internal::PaymentService.new
      end

      def authorize(payment_id, order_id, amount_cents) = @svc.authorize(payment_id, order_id, amount_cents)
      def capture(payment_id) = @svc.capture(payment_id)
      def refund(payment_id) = @svc.refund(payment_id)
    end
  end
end

File: backend-r/R/payments_public_api.R
#' ONLY interface other modules/apps may import from the payments module.
#' @export
payments_authorize <- function(payment_id, order_id, amount_cents) authorize_payment(payment_id, order_id, amount_cents)
#' @export
payments_capture <- function(payment_id) capture_payment(payment_id)
#' @export
payments_refund <- function(payment_id) refund_payment(payment_id)
